includes/wf_crm_webform_preprocess.inc

```diff
@@ -529,7 +529,21 @@ class wf_crm_webform_preprocess extends wf_crm_webform_base {
           // If the user has already entered a value for this field, don't change it
           if (isset($this->info[$ent][$c][$table][$n][$name])
             && !(isset($component['cid']) && isset($submitted[$component['cid']]))) {
-            $val = $this->info[$ent][$c][$table][$n][$name];
+            // pick phone number and email according to selecte location type
+            if (($table == 'phone' && $name == 'phone')
+              || ($table == 'email' && $name == 'email')) {
```

Comment:
why is it only applying to a single field in the entity? Should cover the entire entity I think (this approach seems like it would get the right phone number but the wrong phone extension, for example)